### PR TITLE
added module for HTMX WS extension

### PR DESCRIPTION
**Pull request by h0p3zZ**

### CODE REVIEW

File: Sources/ElementaryHTMXWS/HTMLAttribute+HTMX.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        }
+
+        public static func wss(_ url: String) -> Self {
+            .init(value: "wss:\(url)")
+        }
+    }
+
+    static func connect(_ prefixedURL: WSURLPrefix) -> HTMLAttribute {
+        connect(prefixedURL.value)
+    }
```
Comment (sliemeobn) [inline Sources/ElementaryHTMXWS/HTMLAttribute+HTMX.swift]: this feels a bit too much. isn't just string enough?

**Review by sliemeobn — COMMENTED** [review]
thanks! looking really good.

I'll play around with the demos a bit

File: Sources/ElementaryHTMXWS/HTMLAttribute+HTMX.swift (hunk 1 above)
Comment (h0p3zZ) [reply Sources/ElementaryHTMXWS/HTMLAttribute+HTMX.swift]: Yes, probably true—couldn't decide on whether to add or not to add it.

File: Examples/HummingbirdDemo/Sources/App/Routes.swift [hunk 2]
```diff
@@ -58,3 +60,27 @@ func addRoutes(to router: Router<some RequestContext>) {
...
+            }
+            let frame = WebSocketFrame(
+                fin: frame.fin,
+                opcode: opcode,
+                data: frame.data
+            )
+            try await outbound.write(.text("<div id=\"echo\">Received: \(String(buffer: frame.data))</div>"))
+        }
+    }
+}
```
Comment (sliemeobn) [inline Examples/HummingbirdDemo/Sources/App/Routes.swift]: this feels awfully complicated - I am sure there is a cleaner way somehow

File: Examples/VaporDemo/Sources/App/Routes.swift [hunk 3]
```diff
@@ -29,4 +29,11 @@ func addRoutes(to app: Application) {
             })
         )
     }
+
+    app.webSocket("echo") { req, ws in
+        print(req)
```
Comment (sliemeobn) [inline Examples/VaporDemo/Sources/App/Routes.swift]: good servers don't print ; )

File: Examples/VaporDemo/Sources/App/Views.swift [hunk 4]
```diff
@@ -10,14 +11,21 @@ struct MainPage: HTMLDocument {
...
 
     var body: some HTML {
         header {
             h2 { "Vapor + Elementary + HTMX Demo" }
+            // example of using htmx sse
             div(.hx.ext(.sse), .sse.connect("/time")) {
                 p(.sse.swap("time")) { "Server Time:" }
             }
+            // example of using htmx ws
+            div(.hx.ext(.ws), .ws.connect("/echo"), .hx.vals("{ \"value\": \"TestValue\"}"), .hx.swapOOB(.innerHTML), .hx.target("#echo"), .class("flex justify-between")) {
```
Comment (sliemeobn) [inline Examples/VaporDemo/Sources/App/Views.swift]: for json strings `#"..."#` would be nicer
or ideally have the data come from a text input or something

File: README.md [hunk 5]
```diff
@@ -36,7 +36,6 @@ There is also a `ElementaryHTMXSSE` module that adds the `.sse` syntax for the [
 
 ## Future directions
 
-- Add module for [WebSockets extension](https://github.com/bigskysoftware/htmx-extensions/blob/main/src/ws/README.md)
```
Comment (sliemeobn) [inline README.md:39]: We should add a little chapter to show there is a WS module now as well

File: Sources/ElementaryHTMXWS/HTMLAttribute+HTMX.swift (hunk 1 above)
Comment (sliemeobn) [reply Sources/ElementaryHTMXWS/HTMLAttribute+HTMX.swift]: I think it should go - less API surface

**Review by sliemeobn — CHANGES_REQUESTED** [review]
I think the examples should be a bit cleaner - currently it is visually quite messy and not clear what is happening.

If we do a ping-pong, we could add a name text box or something and have the WS route return a greeting or something similar. maybe a random number generator - something that is not so ugly on the UI ; )

**Comment by sliemeobn** [comment]
lgtm, thanks!

**Pull request merged**